src/gluonts/dataset/parallelized_loader.py

```diff
@@ -346,6 +355,45 @@ def _worker_fn(
     return buf.getvalue()
 
 
+class _ShuffleBatchIter(Iterator[DataEntry]):
+    def __init__(
+        self,
+        base_iterator: Iterator[DataEntry],
+        shuffle_buffer_length: Optional[int],
+        multiprocess: bool,
+    ):
+        if multiprocess:
+            self.shuffle_buffer = _WorkerData.shuffle_buffer
```

Comment:
Why is this needed? Couldn’t you just set `self.shuffle_buffer = []` in any case?